pkg/virt-handler/options_test.go

```diff
@@ -0,0 +1,282 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package virthandler
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	cmdv1 "kubevirt.io/kubevirt/pkg/handler-launcher-com/cmd/v1"
+	"kubevirt.io/kubevirt/pkg/virt-handler/node-labeller/api"
+)
+
+var _ = Describe("Parsing VMI Options", func() {
+	const (
+		memorySize = 2 * 1024 * 1024
+		memoryUnit = "KiB"
+	)
+
+	Context("virt-handler VM processes VMI options during update", func() {
+		It("should convert api capabiliites to cmdv1 topology with single cell", func() {
+			caps := &api.Capabilities{}
+			caps.Host.Topology.Cells = api.Cells{
+				Cell: []api.Cell{
+					{
+						Memory: api.Memory{Unit: memoryUnit, Amount: memorySize},
+						Pages: []api.Pages{
+							{Unit: memoryUnit, Size: 4, Count: 4064224},
+							{Unit: memoryUnit, Size: 2048, Count: 0},
+							{Unit: memoryUnit, Size: 1048576, Count: 0},
+						},
+						Distances: api.Distances{Sibling: []api.Sibling{{ID: 0, Value: 10}}},
+						Cpus: api.CPUs{CPU: []api.CPU{
+							{ID: 0, Siblings: []uint32{0, 4}},
+							{ID: 1, Siblings: []uint32{1, 5}},
+							{ID: 2, Siblings: []uint32{2, 6}},
+							{ID: 3, Siblings: []uint32{3, 7}},
+							{ID: 4, Siblings: []uint32{0, 4}},
+							{ID: 5, Siblings: []uint32{1, 5}},
+							{ID: 6, Siblings: []uint32{2, 6}},
+							{ID: 7, Siblings: []uint32{3, 7}},
+						}},
+					},
+				},
+			}
+
+			actualTopology := capabilitiesToTopology(caps)
+
+			expectedTopology := &cmdv1.Topology{
+				NumaCells: []*cmdv1.Cell{
+					{
+						Memory: &cmdv1.Memory{Unit: memoryUnit, Amount: memorySize},
+						Pages: []*cmdv1.Pages{
+							{Unit: memoryUnit, Size: 4, Count: 4064224},
+							{Unit: memoryUnit, Size: 2048, Count: 0},
+							{Unit: memoryUnit, Size: 1048576, Count: 0},
+						},
+						Distances: []*cmdv1.Sibling{{Id: 0, Value: 10}},
+						Cpus: []*cmdv1.CPU{
+							{Id: 0, Siblings: []uint32{0, 4}},
+							{Id: 1, Siblings: []uint32{1, 5}},
+							{Id: 2, Siblings: []uint32{2, 6}},
+							{Id: 3, Siblings: []uint32{3, 7}},
+							{Id: 4, Siblings: []uint32{0, 4}},
+							{Id: 5, Siblings: []uint32{1, 5}},
+							{Id: 6, Siblings: []uint32{2, 6}},
+							{Id: 7, Siblings: []uint32{3, 7}},
+						},
+					},
+				},
+			}
+
+			Expect(actualTopology).To(Equal(expectedTopology))
+		})
+
+		It("should convert api capabilities to cmdv1 topology with multiple cells, cpus and pages", func() {
+			caps := &api.Capabilities{}
+			caps.Host.Topology.Cells = api.Cells{
+				Cell: []api.Cell{
+					{
+						Memory: api.Memory{Unit: memoryUnit, Amount: memorySize},
```

Comment:
nit: Shouldn't amount here be equal to `1289144` if we try to follow `pkg/virt-handler/node-labeller/testdata/capabilities_with_numa.xml`?